charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml

```
@@ -359,7 +359,7 @@ spec:
 {{- with (omit .Values.prometheus.prometheusSpec.thanos "objectStorageConfig")}}
 {{ toYaml . | indent 4 }}
 {{- end }}
-{{- if .Values.prometheus.prometheusSpec.thanos.objectStorageConfig.existingSecret }}
+{{- if ((.Values.prometheus.prometheusSpec.thanos.objectStorageConfig).existingSecret) }}
```

Thanks for taking the suggestion, but the parenthesis needs to wrap existing fields 😅
```suggestion
{{- if ((.Values.prometheus.prometheusSpec.thanos).objectStorageConfig).existingSecret }}
```

---

`.Values.prometheus.prometheusSpec.thanos` should always exist because it is defaulted to `{}` in values.yaml -- this if block is also wrapped within a check for `.thanos` even if it weren't:

https://github.com/prometheus-community/helm-charts/blob/29eb6a1954a10aa2e4423c0d70aab0c78373cee3/charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml#L357

so I think it's okay to not have to check for existence of `thanos` here?

---

I'd say you still need otherwise it will add an empty config for thanos: https://github.com/prometheus-community/helm-charts/blob/29eb6a1954a10aa2e4423c0d70aab0c78373cee3/charts/kube-prometheus-stack/templates/prometheus/prometheus.yaml#L357-L371

---

Right, I was making an assumption that you either use `.Values.prometheus.prometheusSpec.thanos.objectStorageConfig.existingSecret` or `.Values.prometheus.prometheusSpec.thanos.objectStorageConfig.secret` to define your thanos secrets. But your way is safer.

---

However, you still don't need to wrap the whole key, just up to levels that might be missing.

E.g. `((.Values.aa).bb).cc` when `.Values.aa` or `.Values.aa.bb` might not exist.

---

> However, you still don't need to wrap the whole key, just up to levels that might be missing.

Yes I know, but I personally prefer the style of wrapping the entire scope.

Thanks, I've bumped the chart version and brought it up to date with `main`